src/android/com/microsoftopentechnologies/intellij/helpers/azure/sdk/AzureSDKManagerImpl.java

```diff
@@ -454,6 +826,150 @@ private static DeploymentGetResponse getDeployment(@NotNull ComputeManagementCli
+            throws Exception {
+        StorageAccountGetKeysResponse sagkr = getStorageAccountOperations(client).getKeys(storageName);
+
+        if (sagkr == null) {
+            throw new Exception("Unable to retrieve Storage Account Keys information");
+        }
+
+        return sagkr;
+    }
+
+    @NotNull
+    private static List<Role> getVMDeploymentRoles(@NotNull DeploymentGetResponse deployment) throws Exception {
+        ArrayList<Role> roles = deployment.getRoles();
+
+        if (roles == null) {
+            throw new Exception("Invalid Virtual Machine information. No Roles match the VM data.");
+        }
+
+        return roles;
+    }
+
+    private static void validateOperationStatus(@Nullable OperationStatusResponse osr) throws Exception {
+        if (osr == null) {
+            throw new Exception("Unable to retrieve Operation Status");
+        }
+
+        if (osr.getError() != null) {
+            throw new Exception(osr.getError().getMessage());
+        }
+    }
+
+    @Nullable
+    private static OperationStatusResponse getOperationStatusResponse(@NotNull ComputeManagementClient client,
+                                                                      @NotNull OperationResponse or)
+            throws InterruptedException, java.util.concurrent.ExecutionException, ServiceException {
+        OperationStatusResponse osr = client.getOperationStatusAsync(or.getRequestId()).get();
+        int delayInSeconds = 30;
+
+        if (client.getLongRunningOperationInitialTimeout() >= 0) {
+            delayInSeconds = client.getLongRunningOperationInitialTimeout();
+        }
+
+        while (osr.getStatus() == OperationStatus.InProgress) {
+            Thread.sleep(delayInSeconds * 1000);
+            osr = client.getOperationStatusAsync(or.getRequestId()).get();
+            delayInSeconds = 30;
+
+            if (client.getLongRunningOperationRetryTimeout() >= 0) {
+                delayInSeconds = client.getLongRunningOperationRetryTimeout();
+            }
+        }
+
+        if (osr.getStatus() != OperationStatus.Succeeded) {
+            if (osr.getError() != null) {
+                ServiceException ex = new ServiceException(osr.getError().getCode() + " : " + osr.getError().getMessage());
+                ex.setErrorCode(osr.getError().getCode());
+                ex.setErrorMessage(osr.getError().getMessage());
+                throw ex;
+            } else {
+                throw new ServiceException("");
+            }
+        }
+
+        return osr;
+    }
+
+    @NotNull
+    private static CloudService loadAvailabilitySets(@NotNull ComputeManagementClient client,
+                                                     @NotNull CloudService cloudService)
+            throws Exception {
+        DeploymentGetResponse deployment = getDeployment(client,
+                cloudService.getName(),
+                DeploymentSlot.Production);
+
```

Comment:
This call to `getDeployment` is also made from `getCloudServices`. Maybe we should implement some kind of result caching?